pallets/loans/src/lib.rs

```diff
@@ -1402,4 +1333,46 @@ impl<T: Config> Pallet<T> {
 
         Ok(price)
     }
+
+    // Returns a stored Market.
+    //
+    // Returns `Err` if market does not exist.
+    pub fn market(currency_id: &CurrencyId) -> Result<Market, DispatchError> {
+        Markets::<T>::try_get(currency_id).map_err(|_err| Error::<T>::MarketDoesNotExist.into())
+    }
+
+    // Mutates a stored Market.
+    //
+    // Returns `Err` if market does not exist.
+    pub(crate) fn mutate_market<F>(currency_id: &CurrencyId, cb: F) -> Result<(), DispatchError>
+    where
+        F: FnOnce(&mut Market),
+    {
+        Markets::<T>::try_mutate(currency_id, |opt| {
+            if let Some(market) = opt {
+                cb(market);
+                return Ok(());
+            }
+            Err(Error::<T>::MarketDoesNotExist.into())
+        })
+    }
+
+    // All markets that are `MarketStatus::Active`.
+    fn active_markets() -> impl Iterator<Item = (CurrencyId, Market)> {
+        Markets::<T>::iter().filter(|(_, market)| market.state == MarketState::Active)
+    }
 }
+
+pub const MARKET_MOCK: Market = Market {
```

Comment:
shall we move this struct into mock.rs? I think it won't be used in lib.rs

Reply:
This structure is used by `pallet-loans` and `pallet-loans-benchmarking`. Taking into consideration that `mock.rs` has a `#![cfg(test)]` guard, it is necessary to choose which crate won't use such shared common mock

Reply:
Maybe we could add features to this struct, such as #[cfg(any(test, runtime-benchmarks))] since we don't actually want to compile it to a production binary

Reply:
Done